crates/matrix-sdk-ui/src/room_list_service/filters/non_space.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{super::Room, Filter};
+
+struct NonSpaceRoomMatcher<F>
+where
+    F: Fn(&Room) -> bool,
+{
+    is_not_space: F,
+}
+
+impl<F> NonSpaceRoomMatcher<F>
+where
+    F: Fn(&Room) -> bool,
+{
+    fn matches(&self, room: &Room) -> bool {
+        (self.is_not_space)(room)
+    }
+}
+
+/// Create a new filter that will filter out rooms that are not spaces.
```

Comment:
Should we explain what a space is? Maybe with a link to the spec, like https://spec.matrix.org/latest/client-server-api/#spaces?